sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -412,85 +424,94 @@ async def negotiate(self):
         await self.write(TLS_HEADER_FRAME)
         channel, returned_header = await self.receive_frame(verify_frame_type=None)
         if returned_header[1] == TLS_HEADER_FRAME:
-            raise ValueError("Mismatching TLS header protocol. Excpected: {}, received: {}".format(
-                TLS_HEADER_FRAME, returned_header[1]))
+            raise ValueError(
+                "Mismatching TLS header protocol. Excpected: {}, received: {}".format(
+                    TLS_HEADER_FRAME, returned_header[1]
+                )
+            )
 
 
 class WebSocketTransportAsync(AsyncTransportMixin):
-    def __init__(self, host, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **kwargs
-        ):
+    def __init__(self, host, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **kwargs):
         self._read_buffer = BytesIO()
-        self.loop = get_running_loop()
         self.socket_lock = asyncio.Lock()
-        self.sslopts = ssl if isinstance(ssl, dict) else {}
+        self.sslopts = self._build_ssl_opts(ssl) if isinstance(ssl, dict) else None
         self._connect_timeout = connect_timeout or TIMEOUT_INTERVAL
         self._custom_endpoint = kwargs.get("custom_endpoint")
         self.host = host
         self.ws = None
-        self._http_proxy = kwargs.get('http_proxy', None)
-
+        self.session = None
+        self._http_proxy = kwargs.get("http_proxy", None)
+        self.http_proxy_auth = None
+        self.http_proxy_host = None
+        
     async def connect(self):
-        http_proxy_host, http_proxy_port, http_proxy_auth = None, None, None
+        username, password = None, None
+
         if self._http_proxy:
-            http_proxy_host = self._http_proxy['proxy_hostname']
-            http_proxy_port = self._http_proxy['proxy_port']
-            username = self._http_proxy.get('username', None)
-            password = self._http_proxy.get('password', None)
-            if username or password:
-                http_proxy_auth = (username, password)
+            self.http_proxy_host = self._http_proxy["proxy_hostname"]
+            self.http_proxy_port = self._http_proxy["proxy_port"]
+            if self.http_proxy_host and self.http_proxy_port:
+                self.http_proxy_host += f":{self.http_proxy_port}"
```

Comment:
There doesn't seem to be a reason we are saving proxy host and port to `self`? They're only used in this method right?

Reply:
No particular reason, Ill switch them out to local variables